test/boost/vector_store_client_test.cc

```diff
@@ -459,6 +459,8 @@ SEASTAR_TEST_CASE(vector_store_client_test_ann_request) {
                 auto* err = std::get_if<vector_store_client::service_error>(&keys.error());
                 BOOST_CHECK(err != nullptr);
                 BOOST_CHECK_EQUAL(err->status, status_type::not_found);
+                // expect message as generated by seastar::http::json_exception
+                BOOST_CHECK_EQUAL(err->message, R"({"message": "Not found", "code": 404})");
```

Comment:
This part doesn't really test the logic that you implemented. It just checks that the vector store mock returns the exception in the right format. The `ann_error` -> CQL error conversion logic that you implemented happens on a different abstraction layer, which you are omitting here.

You need to have a test which uses the `cql_test_env` to query an index through the CQL interface, and configure the mock to return some specific error, then match on that error.